src/storage/src/types/sources.rs

```diff
@@ -1489,6 +1506,67 @@ impl RustType<ProtoPubNubSourceConnection> for PubNubSourceConnection {
     }
 }
 
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct LoadGeneratorSourceConnection {
+    pub generator: Generator,
+}
+
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum Generator {
+    Counter,
+}
+
+impl Generator {
+    pub fn data_encoding(&self) -> SourceDataEncoding {
+        let inner = match self {
+            Generator::Counter => DataEncodingInner::RowCodec(
+                RelationDesc::empty().with_column("counter", ScalarType::Int64.nullable(false)),
+            ),
+        };
+        SourceDataEncoding::Single(DataEncoding::new(inner))
+    }
+
+    pub fn by_offset(&self, offset: MzOffset) -> Row {
```

Comment:
This is cool because it's deterministic but maybe that's bad too because things should be random? Unsure.

Reply:
I think deterministic is great. Randomness could be recovered by adding a `SEED` option to the load generators that use a PRNG. Users who really want random data could run `CREATE SOURCE ... LOAD GENERATOR tpch_users SEED now()` or something like that.